Replace crypto/derive-key-for-all-supported-algorithms.html with a WPT test

javifernandez opened the pull request:
#### 1e03134fca748e5a92df05a02242f4471755787e
<pre>
Replace crypto/derive-key-for-all-supported-algorithms.html with a WPT test
<a href="https://bugs.webkit.org/show_bug.cgi?id=282285">https://bugs.webkit.org/show_bug.cgi?id=282285</a>

Reviewed by Sihui Liu.

Upstream commit: <a href="https://github.com/web-platform-tests/wpt/commit/484258ee19babc648526001ecf415717e360c868">https://github.com/web-platform-tests/wpt/commit/484258ee19babc648526001ecf415717e360c868</a>

* LayoutTests/crypto/subtle/derive-key-for-all-supported-algorithms-expected.txt: Removed.
* LayoutTests/crypto/subtle/derive-key-for-all-supported-algorithms.html: Removed.
* LayoutTests/imported/w3c/web-platform-tests/WebCryptoAPI/derive_bits_keys/derive_key_and_encrypt.https.any.html: Added.
* LayoutTests/imported/w3c/web-platform-tests/WebCryptoAPI/derive_bits_keys/derive_key_and_encrypt.https.any.js: Added.
* LayoutTests/imported/w3c/web-platform-tests/WebCryptoAPI/derive_bits_keys/derive_key_and_encrypt.https.any.worker.html: Added.
* LayoutTests/imported/w3c/web-platform-tests/WebCryptoAPI/derive_bits_keys/derive_key_and_encrypt.js: Added.
(async importKeys):
* LayoutTests/imported/w3c/web-platform-tests/WebCryptoAPI/derive_bits_keys/w3c-import.log:

Canonical link: <a href="https://commits.webkit.org/285896@main">https://commits.webkit.org/285896@main</a>
</pre>
<!--EWS-Status-Bubble-Start-->
https://github.com/WebKit/WebKit/commit/6da2ad630b13e04293173470eb86e667d1a0177f

| Misc | iOS, visionOS, tvOS & watchOS  | macOS  | Linux |  Windows |
| ----- | ---------------------- | ------- |  ----- |  --------- |
| [✅ 🧪 style](https://ews-build.webkit.org/#/builders/38/builds/74059 "Passed style check") | [✅ 🛠 ios](https://ews-build.webkit.org/#/builders/48/builds/53488 "Built successfully") | [✅ 🛠 mac](https://ews-build.webkit.org/#/builders/55/builds/26870 "Built successfully") | [✅ 🛠 wpe](https://ews-build.webkit.org/#/builders/5/builds/78465 "Built successfully") | [✅ 🛠 win](https://ews-build.webkit.org/#/builders/59/builds/25297 "Built successfully") 
| | [✅ 🛠 ios-sim](https://ews-build.webkit.org/#/builders/49/builds/62621 "Built successfully") | [✅ 🛠 mac-AS-debug](https://ews-build.webkit.org/#/builders/123/builds/1273 "Built successfully") | [✅ 🧪 wpe-wk2](https://ews-build.webkit.org/#/builders/34/builds/58251 "Passed tests") | [✅ 🧪 win-tests](https://ews-build.webkit.org/#/builders/60/builds/16570 "Passed tests") 
| [✅ 🧪 webkitperl](https://ews-build.webkit.org/#/builders/11/builds/77126 "Passed tests") | [✅ 🧪 ios-wk2](https://ews-build.webkit.org/#/builders/47/builds/48377 "Passed tests") | [✅ 🧪 api-mac](https://ews-build.webkit.org/#/builders/18/builds/63701 "Passed tests") | [✅ 🧪 api-wpe](https://ews-build.webkit.org/#/builders/41/builds/38661 "Passed tests") | 
| | [✅ 🧪 ios-wk2-wpt](https://ews-build.webkit.org/#/builders/42/builds/45243 "Passed tests") | [✅ 🧪 mac-wk1](https://ews-build.webkit.org/#/builders/64/builds/21193 "Passed tests") | [✅ 🛠 wpe-cairo](https://ews-build.webkit.org/#/builders/65/builds/23630 "Built successfully") | 
| | [✅ 🧪 api-ios](https://ews-build.webkit.org/#/builders/13/builds/66754 "Passed tests") | [✅ 🧪 mac-wk2](https://ews-build.webkit.org/#/builders/63/builds/21540 "Passed tests") | [✅ 🛠 gtk](https://ews-build.webkit.org/#/builders/2/builds/79951 "Built successfully") | 
| | [✅ 🛠 vision](https://ews-build.webkit.org/#/builders/128/builds/1376 "Built successfully") | [✅ 🧪 mac-AS-debug-wk2](https://ews-build.webkit.org/#/builders/122/builds/751 "Passed tests") | [✅ 🧪 gtk-wk2](https://ews-build.webkit.org/#/builders/1/builds/66534 "Passed tests") | 
| | [✅ 🛠 vision-sim](https://ews-build.webkit.org/#/builders/121/builds/1520 "Built successfully") | [✅ 🧪 mac-wk2-stress](https://ews-build.webkit.org/#/builders/8/builds/63716 "Passed tests") | [❌ 🧪 api-gtk](https://ews-build.webkit.org/#/builders/21/builds/65808 "Found 1 new API test failure: /WebKitGTK/TestWebsiteData:/webkit/WebKitWebsiteData/service-worker-registrations (failure)") | 
| [✅ 🛠 🧪 merge](https://ews-build.webkit.org/#/builders/19/builds/16327 "Built successfully and passed tests") | [✅ 🧪 vision-wk2](https://ews-build.webkit.org/#/builders/126/builds/9721 "Passed tests") | [✅ 🧪 mac-intel-wk2](https://ews-build.webkit.org/#/builders/119/builds/7895 "Passed tests") | | 
| | [✅ 🛠 tv](https://ews-build.webkit.org/#/builders/127/builds/1340 "Built successfully") | | | 
| | [✅ 🛠 tv-sim](https://ews-build.webkit.org/#/builders/125/builds/1369 "Built successfully") | | | 
| | [✅ 🛠 watch](https://ews-build.webkit.org/#/builders/129/builds/1357 "Built successfully") | | | 
| | [✅ 🛠 watch-sim](https://ews-build.webkit.org/#/builders/124/builds/1376 "Built successfully") | | | 
<!--EWS-Status-Bubble-End-->

nmahendru commented on LayoutTests/imported/w3c/web-platform-tests/WebCryptoAPI/derive_bits_keys/derive_key_and_encrypt.js:
```diff
@@ -0,0 +1,49 @@
+}
+
+async function importKeys() {
+    // "ECDSA" with a 'P-256' curve
```
It is probably helpful to add the stub code that I had to generate this data. It's useful to quickly use when adding more tests or debugging by writing a new test.

javifernandez replied:
I'll submit a new PR later to add it.

nmahendru reviewed (approved):
Thanks for moving this over!

javifernandez committed:
Replace crypto/derive-key-for-all-supported-algorithms.html with a WPT test
https://bugs.webkit.org/show_bug.cgi?id=282285

Reviewed by Sihui Liu.

Upstream commit: https://github.com/web-platform-tests/wpt/commit/484258ee19babc648526001ecf415717e360c868

* LayoutTests/crypto/subtle/derive-key-for-all-supported-algorithms-expected.txt: Removed.
* LayoutTests/crypto/subtle/derive-key-for-all-supported-algorithms.html: Removed.
* LayoutTests/imported/w3c/web-platform-tests/WebCryptoAPI/derive_bits_keys/derive_key_and_encrypt.https.any.html: Added.
* LayoutTests/imported/w3c/web-platform-tests/WebCryptoAPI/derive_bits_keys/derive_key_and_encrypt.https.any.js: Added.
* LayoutTests/imported/w3c/web-platform-tests/WebCryptoAPI/derive_bits_keys/derive_key_and_encrypt.https.any.worker.html: Added.
* LayoutTests/imported/w3c/web-platform-tests/WebCryptoAPI/derive_bits_keys/derive_key_and_encrypt.js: Added.
(async importKeys):
* LayoutTests/imported/w3c/web-platform-tests/WebCryptoAPI/derive_bits_keys/w3c-import.log:

Canonical link: https://commits.webkit.org/285896@main
